### PR TITLE
Unify ram buckets

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

**Comment by kasperl** [comment]
This looks like a big change, but it really is just a refactoring that splits the storage implementation into a couple of files and unifies the functionality, so we use the same code for device and host targets. 

On host targets, we now simulate the RTC memory using a static array. This means that we can test the (slightly broken) implementation of RAM buckets more easily.

File: system/storage/region.toit [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    revoke
+
+  static open provider/StorageServiceProvider client/int -> List
+      --path/string
+      --capacity/int?:
+    registry := provider.registry
+    id := uuid.uuid5 NAMESPACE path
+    allocation := find_allocation_ registry --id=id --if_absent=:
+      if not capacity: throw "FILE_NOT_FOUND"
+      new_allocation_ registry --id=id --path=path --size=capacity + FLASH_REGISTRY_PAGE_SIZE
```
Comment (floitsch) [inline system/storage/region.toit:57]: Add a comment explaining why we need the extra FLASH_REGISTRY_PAGE_SIZE

File: system/storage/region.toit [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        Region.MODE_WRITE_CAN_CLEAR_BITS_
+    ]
+
+  static delete registry/FlashRegistry -> none
+      --path/string:
+    id := uuid.uuid5 NAMESPACE path
+    allocation := find_allocation_ registry --id=id --if_absent=: return
+    offset := allocation.offset + FLASH_REGISTRY_PAGE_SIZE
+    size := allocation.size - FLASH_REGISTRY_PAGE_SIZE
+    if flash_is_accessed_ offset size: throw "ALREADY_IN_USE"
```
Comment (floitsch) [inline system/storage/region.toit:76]: `STILL_IN_USE`, or something similar sounds more correct?

**Review by floitsch — APPROVED** [review]
LGTM.

File: system/storage/region.toit (hunk 2 above)
Comment (kasperl) [reply system/storage/region.toit:76]: ALREADY_IN_USE is a "standard" exception we generate from primitives too. I think I prefer sticking to that.

File: system/storage/region.toit (hunk 1 above)
Comment (kasperl) [reply system/storage/region.toit:60]: Done.

**Pull request merged**